src/main/java/com/mojang/brigadier/CommandDispatcher.java

```diff
@@ -391,16 +393,23 @@ private ParseResults<S> parseNodes(final CommandNode<S> node, final StringReader
                     return new ParseResults<>(context, parse.getReader(), parse.getExceptions());
                 } else {
                     final ParseResults<S> parse = parseNodes(child, reader, context);
+                    if (unique) {
+                        return parse;
+                    }
```

Comment:
If `unique`, the `potentials` will always contain only `parse` and this `parse` will be returned after exiting this `for` loop, so we can return `parse` here without allocating `potentials`.